src/core/load-fragment.c

```diff
@@ -6575,3 +6577,62 @@ int config_parse_tty_size(
 
         return config_parse_unsigned(unit, filename, line, section, section_line, lvalue, ltype, rvalue, data, userdata);
 }
+
+int config_parse_log_filter_patterns(
+                const char *unit,
+                const char *filename,
+                unsigned line,
+                const char *section,
+                unsigned section_line,
+                const char *lvalue,
+                int ltype,
+                const char *rvalue,
+                void *data,
+                void *userdata) {
+
+        Set **patterns = ASSERT_PTR(data);
+        const char *new_pattern = ASSERT_PTR(rvalue);
+        _cleanup_(pattern_freep) pcre2_code *compiled_pattern = NULL;
+        int r;
+
+        assert(filename);
+        assert(lvalue);
+
+        if (isempty(new_pattern)) {
+                /* Empty assignment resets the list. */
+                *patterns = set_free_free(*patterns);
+                return 0;
+        }
+
+        if (new_pattern[0] == '~') {
+                /* If LogFilterPatterns starts with ~, we skip it to test compile the pattern. */
+                new_pattern++;
+
+                if (new_pattern[0] == '\0') {
+                        log_syntax(unit, LOG_WARNING, filename, line, -EINVAL,
```

Comment:
passing -EINVAL is pointless, if you are making the error up here and do not to intend the return value of log_syntax(). Please pass 0 instead.